### ISSUE TITLE
Function dataflow becomes stale after re-analysis

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#1249]
From an object I wrote during an episode of my stream:

```python
class UnlockTaskThread(BackgroundTaskThread):
    def __init__(self, function, addr):
        super(UnlockTaskThread, self).__init__()
        self.addr = addr
        self.function = function
        self.view = function.view
```
In the `run` method of this class, I loop for a while and patch the function referenced by `self.function`, which adds new basic blocks. If I try to call `self.function.get_reg_value_at(addr, 'esp')` after a `view.update_analysis_and_wait()`, I will get back `<undetermined>` but if I do something like `self.function = view.get_functions_containing(addr)[0]` after updating the analysis and then check the value of the register, I'll get `<stack frame offset -0x4>`

I'm not sure if in the core my Function's backing object is being replaced by a new Function object but I'm still holding a reference to the previous one, or what....but I think the expectation here should be that a python Function object should always be up to date with the core Function object after updating analysis.

**Comment by plafosse** [comment]
Possible dupe of #1221

**Comment by joshwatson** [comment]
Yeah, I wasn't exactly sure since it sounded a little different, particularly in that the other one persisted even when you deleted and recreated a function
